pkg/kubelet/util/env/env_util_test.go

```diff
@@ -36,245 +68,483 @@ func TestParseEnv(t *testing.T) {
 	tests := []testCase{
 		{
 			name: "ignore leading whitespace",
-			envContent: `   KEY1=val1
-		KEY2=val2
-KEY3=val3
+			envContent: `   KEY1='val1'
+		KEY2='val2'
+KEY3='val3'
 `,
 			key:       "KEY2",
-			wantValue: "val2",
+			wantValue: `val2`,
 		},
 		{
 			name: "ignore blank and comment lines",
 			envContent: `# comment
 
-KEY1=foo
+KEY1='foo'
    # another comment
-	
-KEY2=bar
+
+KEY2='bar'
 `,
 			key:       "KEY2",
-			wantValue: "bar",
+			wantValue: `bar`,
+		},
+		{
+			name: "whitespace around = and not allowed",
+			envContent: `KEY1 = 'val1'
+KEY2=   'val2'
+KEY3='val3'
+`,
+			key:         "KEY2",
+			wantErr:     true,
+			errContains: `whitespace before '=' is not allowed`,
+		},
+		{
+			name: "key not found returns empty",
+			envContent: `KEY1='foo'
+KEY2='bar'
+`,
+			key:       "KEY3",
+			wantValue: ``,
+		},
+		{
+			name: "value with embedded #",
+			envContent: `KEY1='foo#notcomment'
+KEY2='bar'
+`,
+			key:       "KEY1",
+			wantValue: `foo#notcomment`,
+		},
+		{
+			name: "key with whitespace (should error)",
+			envContent: `KEY1 ='foo'
+KEY2='bar'
+`,
+			key:         "KEY1",
+			wantErr:     true,
+			errContains: "whitespace before '=' is not allowed",
 		},
 		{
-			name: "whitespace around = and trailing",
-			envContent: `KEY1 = val1   
-KEY2=   val2	
-KEY3=val3   	
+			name: "value with leading and trailing whitespace",
+			envContent: `KEY1='   foo bar   '
+KEY2='bar'
+`,
+			key:       "KEY1",
+			wantValue: `   foo bar   `,
+		},
+		{
+			name: "multiple comments and blank lines",
+			envContent: `# first comment
+
+# second comment
+KEY1='foo'
+
+# third comment
+KEY2='bar'
 `,
 			key:       "KEY2",
-			wantValue: "val2",
+			wantValue: `bar`,
 		},
 		{
-			name: "continuation line with \\",
-			envContent: `KEY1=foo \
-bar \
-baz
-KEY2=val2
+			name: "empty value",
+			envContent: `KEY1='foo'
+KEY2=''
+KEY3='bar'
+`,
+			key:       "KEY2",
+			wantValue: ``,
+		},
+		{
+			name: "key is empty (should error)",
+			envContent: `='foo'
+KEY2='bar'
+`,
+			key:         "KEY2",
+			wantValue:   `bar`,
```

Comment:
super nit: Since we’re expecting an error, I don’t think `wantValue` is necessary here.

```suggestion
```